File: editorial/chap03/cpp/answer_B11.cpp
#include <iostream>
#include <algorithm>
using namespace std;
 
int N, A[100009];
int Q, X[100009];
 
int main() {
	// 입력
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> A[i];
	cin >> Q;
	for (int i = 1; i <= Q; i++) cin >> X[i];
 
	// 배열 X 정렬
	sort(A + 1, A + N + 1);
 
	// 질문에 답한다
	for (int i = 1; i <= Q; i++) {
		int pos1 = lower_bound(A + 1, A + N + 1, X[i]) - A;
		cout << pos1 - 1 << endl;
	}
	return 0;
}

File: editorial/chap03/python/answer_B11.py
import bisect

# 입력
N = int(input())
A = list(map(int, input().split()))
Q = int(input())
X = [ None ] * Q
for i in range(Q):
	X[i] = int(input())

# 배열 X 정렬
A.sort()

# 질문에 답한다(배열 A는 0번부터 시작한다는 점에 유의하세요!)
for i in range(Q):
	pos1 = bisect.bisect_left(A, X[i])
	print(pos1)
